fuzzamoto-ir/src/compiler.rs

```diff
@@ -1500,6 +1536,29 @@ impl Compiler {
                 self.connection_counter += 1;
                 self.append_variable(connection_id);
             }
+            Operation::AddConnectionWithHandshake => {
+                let node_var = self.get_input::<usize>(&instruction.inputs, 0)?;
+                let connection_type_var = self.get_input::<String>(&instruction.inputs, 1)?;
+                let handshake_opts_var = self.get_input::<HandshakeOpts>(&instruction.inputs, 2)?;
+                let time_var = self.get_input::<u64>(&instruction.inputs, 3)?;
+
+                self.output
+                    .actions
+                    .push(CompiledAction::ConnectAndHandshake {
+                        node: *node_var,
+                        connection_type: connection_type_var.clone(),
+                        relay: handshake_opts_var.relay,
+                        starting_height: handshake_opts_var.starting_height,
+                        wtxidrelay: handshake_opts_var.wtxidrelay,
+                        addrv2: handshake_opts_var.addrv2,
+                        erlay: handshake_opts_var.erlay,
+                        time: *time_var,
+                    });
+
+                let connection_id = self.connection_counter;
+                self.connection_counter += 1;
```

Comment:
Just curious, what happens if the handshake options are incompatible or if the connection dies? Does the compiler make a program that won't execute?